python/pyspark/sql/session.py

```diff
@@ -1023,6 +1023,12 @@ def prepare(obj: Any) -> Any:
 
         if isinstance(data, RDD):
             rdd, struct = self._createFromRDD(data.map(prepare), schema, samplingRatio)
+        elif isinstance(data, list) and schema is None:
```

Comment:
Respect the existing behavior when `schema` is given.

Specifically, 
```py
>>> spark.createDataFrame([1, 2], schema='int')
DataFrame[value: int]

>>> spark.createDataFrame([(1,), (2,)], schema='int')
...
TypeError: field value: IntegerType() can not accept object (1,) in type <class 'tuple'>
```

Adjusting the inconsistency above is out of the scope of this PR and ought to be a follow-up.